server/test/services/applications/ProgramAdminApplicationServiceTest.java

```diff
@@ -105,4 +164,192 @@ public void getNote_multipleNotes_findsLatest() throws Exception {
+
+    verify(simpleEmail, times(1))
+        .send(
+            eq(tiEmail),
+            Mockito.contains("An update on the application for program"),
+            Mockito.contains("has changed to " + STATUS_WITH_ONLY_ENGLISH_EMAIL.statusText()));
+    verify(simpleEmail, times(1))
+        .send(
+            eq(userEmail),
+            eq(String.format(service.STATUS_UPDATE_EMAIL_SUBJECT_FORMAT, programName)),
+            Mockito.contains(
+                STATUS_WITH_ONLY_ENGLISH_EMAIL.localizedEmailBodyText().get().getDefault()));
+  }
+
+  @Test
+  public void setStatus_invalidStatus_throws() throws Exception {
+    String userEmail = "user@email.com";
+
+    ProgramDefinition program =
+        ProgramBuilder.newActiveProgram("some-program")
+            .withStatusDefinitions(new StatusDefinitions(ORIGINAL_STATUSES))
+            .buildDefinition();
+    Account account = resourceCreator.insertAccount();
+    Applicant applicant = resourceCreator.insertApplicantWithAccount(Optional.of(userEmail));
+    Application application =
+        Application.create(applicant, program.toProgram(), LifecycleStage.ACTIVE)
+            .setSubmitTimeToNow();
+
+    StatusEvent event =
+        StatusEvent.builder().setEmailSent(true).setStatusText("Not an actual status").build();
+
+    assertThrows(
+        StatusNotFoundException.class, () -> service.setStatus(application, event, account));
+  }
+
+  @Test
+  @Parameters({
+    // Status does not have email.
+    "STATUS_WITH_NO_EMAIL,true,user@email.com",
+    // Admin doesn't request an email be sent.
+    "STATUS_WITH_ONLY_ENGLISH_EMAIL,false,user@email.com",
+    // User does no have an email.
+    "STATUS_WITH_ONLY_ENGLISH_EMAIL,true,",
+  })
+  public void setStatus_doesNotSendEmail(
+      String status, boolean adminRequestsEmail, @Nullable String userEmail) throws Exception {
+    SimpleEmail simpleEmail = Mockito.mock(SimpleEmail.class);
+    service =
+        new ProgramAdminApplicationService(
+            instanceOf(ApplicantService.class),
+            instanceOf(ApplicationRepository.class),
+            instanceOf(ApplicationEventRepository.class),
+            instanceOf(Config.class),
+            simpleEmail);
+
+    ProgramDefinition program =
+        ProgramBuilder.newActiveProgram("some-program")
+            .withStatusDefinitions(new StatusDefinitions(ORIGINAL_STATUSES))
+            .buildDefinition();
+    Account account = resourceCreator.insertAccount();
+    Applicant applicant =
+        resourceCreator.insertApplicantWithAccount(Optional.ofNullable(userEmail));
+    Application application =
+        Application.create(applicant, program.toProgram(), LifecycleStage.ACTIVE)
+            .setSubmitTimeToNow();
+
+    StatusEvent event =
+        StatusEvent.builder().setEmailSent(adminRequestsEmail).setStatusText(status).build();
+
+    service.setStatus(application, event, account);
+
+    verify(simpleEmail, never()).send(anyString(), anyString(), anyString());
```

Comment:
Per my comment in the service class, can we add an assertion of the logged event, specifically for the (hopefully) degenerate cases of admin/user requesting an email is sent for either 1) an application with no submitter email (e.g. guest) and 2) no email is configured for the status definition.

In my opinion, situation #2 would be a runtime error and situation #1 would have a logged status event indicating no email was sent.